apps/engine/lib/engine/block.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Engine.Block do
+  @moduledoc """
+  Represent a block of transactions that will be submitted to the contracts.
+  """
+
+  use Ecto.Schema
+
+  import Ecto.Changeset
+  import Ecto.Query, only: [from: 2]
+
+  schema "blocks" do
+    field(:hash, :binary)
+    field(:number, :integer)
+
+    has_many(:transactions, Engine.Transaction)
+
+    timestamps(type: :utc_datetime)
+  end
+
+  @doc """
+  Forms a pending block record based on the existing pending transactions. This
+  attaches free transactions into a new block, awaiting for submission to the contract
+  later on.
+  """
+  @spec form_block() :: {non_neg_integer, non_neg_integer}
+  def form_block() do
```

Comment:
Ah I should preface this in the main thread, but this is still a draft so there's still rough edges (I made up the 25 limit). 

1. Yes, the initial intention is to do that. The assumption here is to match parity with the engine in rocksdb, perhaps the way to do it would be to match the creation of transaction with the main engine, and when `OMG.State.form_block` is called, we could assume that all pending/free transactions are rolled up into the same block. Re: limit I made up and I can remove this.

2. I'm open to a state machine here. I was not sure what other states we would want to consider but going this route long term would be a nice approach. For the current case, this seems to suffice.

3. Good catch. We can make this explict by ordering  by the `inserted_at` timestamp in the query, will add that in.

Reply:
OK thanks, this clarifies this somewhat. To follow-up
1. ah sorry, I might have misspoke with the "attach" _some_ free txs part. To simplify the question: Is `Engine.block.form_block` reflecting `OMG.State.form_block`, where _all_ pending txs are included or is `Engine.Block.form_block` a new concept, where _only some_ free txs are scheduled for a block. And we call the latter a few times per every block formed (in the old sense) and sealed?
2. Aha, I'd slightly prefer (even a trivial) state machine to `is_nil`, for the sake of readability, from day one. Or, instead of the trivial SM, there could be a dedicated relation like `Engine.ScheduledInclusions` (made up name), that could then be checked for `is_nil` more explicilty for the purpose of this process. No strong opinion here though.
3. Hm, be careful with this. This ties to the reply we're giving to the tx submitter (which includes txindex), tx chaining and tx semantics. You need to be extra sure that you won't rearrange the txs inside the block in an incorrect way.

EDIT: I just read the other comment :point_down:, so feel free to ignore the above, as they might tie strongly to the existing implementation and behaviors.